pkg/testexecutionservice/testexecution.go

```diff
@@ -215,21 +231,78 @@ func (tes *testExecutionService) buildCmdArgs(ctx context.Context,
 	frameWork string,
 	frameworkVersion int,
 	payload *core.Payload,
-	target []string) ([]string, error) {
+	target []string) ([]string, string, error) {
 	args := []string{global.FrameworkRunnerMap[frameWork]}
 
 	args = append(args, utils.GetArgs("execute", frameWork, frameworkVersion, testConfigFile, target)...)
 
+	locatorFilePath := ""
 	if payload.LocatorAddress != "" {
 		locatorFile, err := tes.getLocatorsFile(ctx, payload.LocatorAddress)
 		tes.logger.Debugf("locators : %v\n", locatorFile)
 		if err != nil {
 			tes.logger.Errorf("failed to get locator file, error: %v", err)
-			return nil, err
+			return nil, "", err
 		}
 
 		args = append(args, global.ArgLocator, locatorFile)
+		locatorFilePath = locatorFile
+	}
+
+	return args, locatorFilePath, nil
+}
+
+// Read locators from the file and convert it into array of locator config
+func extractLocators(locatorFilePath, flakyTestAlgo string, logger lumber.Logger) ([]core.LocatorConfig, error) {
+	locatorArrTemp := []core.LocatorConfig{}
+	inputLocatorConfigTemp := &core.InputLocatorConfig{}
+
+	if flakyTestAlgo == core.RunningXTimesShuffle {
+		content, err := os.ReadFile(locatorFilePath)
+		if err != nil {
+			logger.Errorf("Error when opening file: ", err)
+			return nil, err
+		}
+
+		err = json.Unmarshal(content, &inputLocatorConfigTemp)
+		if err != nil {
+			logger.Errorf("Error during Unmarshal(): ", err)
+			return nil, err
+		}
+		locatorArrTemp = inputLocatorConfigTemp.Locators
 	}
 
-	return args, nil
+	return locatorArrTemp, nil
+}
+
+// shuffling order of elements in locator array
+func shuffleLocators(locatorArr []core.LocatorConfig, locatorFilePath string, logger lumber.Logger) error {
```

Comment:
too much logic in test execution. Why test-execution has a **direct** responsibility for shuffling? Refactor it out in a different pkg and define the functions there.